File: Paulo_Code/Prototipo_java_single-thread/src/main/java/br/pucminas/grafo/core/Application.java
package br.pucminas.grafo.core;

import br.pucminas.grafo.events.EventBus;
import br.pucminas.grafo.graph.mining.MiningHandler;
import br.pucminas.grafo.handlers.*;

/**
 * Raiz de composição da aplicação.
 *
 * <p>
 * Instancia todos os componentes, registra os handlers no EventBus
 * e expõe o bus para uso externo (CLI, testes, etc.).
 * </p>
 */
public class Application {

    private final EventBus bus;
    private final GraphRegistry registry;

    // handlers (mantidos como campos para possível re-registro futuro)
    private final GraphHandler graphHandler;
    private final AlgorithmHandler algorithmHandler;
    private final MetricsHandler metricsHandler;
    private final MiningHandler miningHandler;

    public Application() {
        this.registry = new GraphRegistry();
        this.bus = new EventBus();

        this.graphHandler = new GraphHandler(registry);
        this.algorithmHandler = new AlgorithmHandler(registry);
        this.metricsHandler = new MetricsHandler(registry);
        this.miningHandler = new MiningHandler(registry);

        // Registra todos os handlers no bus
        graphHandler.registerAll(bus);
        algorithmHandler.registerAll(bus);
        metricsHandler.registerAll(bus);
        miningHandler.registerAll(bus);
    }

    public EventBus getBus() {
        return bus;
    }

    public GraphRegistry getRegistry() {
        return registry;
    }
}

File: Projeto_Python_Apendice/grafo/core/application.py
from .graph_registry import GraphRegistry
from ..events.event_bus import EventBus
from ..handlers.graph_handler import GraphHandler
from ..handlers.algorithm_handler import AlgorithmHandler
from ..handlers.metrics_handler import MetricsHandler
from ..handlers.mining_handler import MiningHandler

class Application:
    def __init__(self):
        self.registry = GraphRegistry()
        self.bus = EventBus()
        
        self.graph_handler = GraphHandler(self.registry)
        self.algo_handler = AlgorithmHandler(self.registry)
        self.metrics_handler = MetricsHandler(self.registry)
        self.mining_handler = MiningHandler(self.registry)
        
        self.graph_handler.register_all(self.bus)
        self.algo_handler.register_all(self.bus)
        self.metrics_handler.register_all(self.bus)
        self.mining_handler.register_all(self.bus)

    def get_bus(self) -> EventBus:
        return self.bus

    def get_registry(self) -> GraphRegistry:
        return self.registry
